dev/AppNotifications/AppNotificationContentBuilder/AppNotificationContentBuilder.idl

```diff
@@ -0,0 +1,138 @@
+
+    // Adds a key / value to the activation arguments that will be returned when the toast notification or its buttons are clicked.
+    AppNotificationContentBuilder AddArgument(String, Enum)
+
+    // Adds a key / value to the activation arguments that will be returned when the toast notification or its buttons are clicked.
+    AppNotificationContentBuilderAddArgument(String, Int32)
+
+    // Adds a key / value to the activation arguments that will be returned when the toast notification or its buttons are clicked.
+    AppNotificationContentBuilder AddArgument(String, Single)
+
+    // Adds a key / value to the activation arguments that will be returned when the toast notification or its buttons are clicked.
+    AppNotificationContentBuilder AddArgument(String, String)
+
+    // Add an Attribution Text to be displayed on the toast.
+    AppNotificationContentBuilder AddAttributionText(String)
+
+    // Add an Attribution Text to be displayed on the toast.
+    AppNotificationContentBuilder AddAttributionText(String, String)
+
+    // Add a button to the current toast.
+    AppNotificationContentBuilder AddButton(IToastButton)
+
+    // Add an button to the toast that will be display to the right of the input text box, achieving a quick reply scenario.
+    AppNotificationContentBuilder AddButton(String, String, ToastActivationType, String)
+
+    // Add an button to the toast that will be display to the right of the input text box, achieving a quick reply scenario.
+    AppNotificationContentBuilder AddButton(String, String, ToastActivationType, String, Uri)
+
+    // Add a button to the current toast.
+    AppNotificationContentBuilder AddButton(String, ToastActivationType, String)
+
+    // Add a button to the current toast.
+    AppNotificationContentBuilder AddButton(String, ToastActivationType, String, Uri)
+
+    // Add an image inline with other toast content.
+    AppNotificationContentBuilder AddInlineImage(Uri, String, Nullable<Boolean>, Nullable<AdaptiveImageCrop>, Nullable<Boolean>)
+
+    // Add an input text box that the user can type into.
+    AppNotificationContentBuilder AddInputTextBox(String, String, String)
+
+    // Add text to the toast.
+    AppNotificationContentBuilder AddText(String, Nullable<AdaptiveTextStyle>, Nullable<Boolean>, Nullable<Int32>, Nullable<Int32>, Nullable<AdaptiveTextAlign>, String)
+
+    // Add an input option to the Toast.
+    AppNotificationContentBuilder AddToastInput(IToastInput)
+
+    // Get the instance of ToastContent that has been built by the builder with specified configuration so far.
+    AppNotificationContentBuilder GetToastContent()
+
+    // Retrieves the notification XML content as a WinRT XmlDocument, so that it can be used with a local Toast notification's constructor on either Windows.UI.Notifications.ToastNotification or Windows.UI.Notifications.ScheduledToastNotification.
+    AppNotificationContentBuilder GetXml()
```

Comment:
I see pushback with using XmlDocument due to the pushback from the last API review. We'd want a GetXml that returns a string, right? @hulumane

Reply:
The pushback was against using Xml DOM. We should use plain string instead.
Also this pattern looks strange here. How does it work? Isn't GetXml supposed to return an XML string and not a Builder object?
I can imagine SetXml(String xml) returning an AppBuilder but not this call.